hnclient-v2/src/main/java/ca/bc/gov/hlth/hnclientv2/handshakeserver/ConnectionHandler.java

```diff
@@ -0,0 +1,432 @@
+	
+	// message header indicator
+	public static final String HEADER_INDICATOR = "MSH";
+	
+	// data indicator length
+	private static final int DATA_INDICATOR_LENGTH = 2;
+	
+	// Segment length for both DT and SI messages
+	private static final int SEGMENT_LENGTH = 12;
+	
+	// Station Identifier length
+	private static final int SI_MESSAGE_LENGTH = 44;	
+	
+	// Hand shake segment
+	private static final String HS_SEGMENT ="HS0000000008"; 
+		
+	// Header padding
+	private static final String TEN_ZEROS = "0000000000";
+		
+	// DT segments to send to POS
+	private byte[] dataSegmentOut = new byte[SEGMENT_LENGTH];
+	private byte[] dataHL7out;
+
+	private byte decodeSeed = 0;
+	
+	private String hnSecureResponse = "";
+	
+    /** Time in ms to wait between reads */
+    private Integer socketReadSleepTime;
+
+    /** Number of reads to attempt */
+    private Integer maxSocketReadTries;
+   
+	private final Socket socket;
+	
+	private final ProducerTemplate producer;
+	
+	public ConnectionHandler(ProducerTemplate producer, Socket socket, Integer socketReadSleepTime, Integer maxSocketReadTries) {
+		this.producer = producer;
+		this.socket = socket;
+		this.socketReadSleepTime = socketReadSleepTime;
+		this.maxSocketReadTries = maxSocketReadTries;
+	}
+
+	@Override
+	public Void call() throws Exception {
+		BufferedInputStream socketInput = null;
+		BufferedOutputStream socketOutput = null;
+		String ret_code = null;
+		try {
+			logger.info("{} - Accepting connection attempt from IP Address: {}", LoggingUtil.getMethodName(),
+					socket.getInetAddress().getHostAddress());
+
+			socketInput = new BufferedInputStream(socket.getInputStream(), 1000);
+			socketOutput = new BufferedOutputStream(socket.getOutputStream());
+
+			ret_code = xfer_ReceiveHSSegment(socketOutput, socketInput, XFER_HANDSHAKE_SEED);
+
+			if (ret_code.contentEquals(MessageUtil.HNET_RTRN_SUCCESS)) {
+				performTransaction(socketInput, socketOutput);
+			} else {
+				logger.warn("{} - Handshake failed: {} {}", LoggingUtil.getMethodName(), ret_code, System.lineSeparator());
+				hnSecureResponse = ErrorBuilder.buildDefaultErrorMessage(ret_code);
+				sendResponse(socketOutput, hnSecureResponse);
+				// reset decodeseed
+				decodeSeed = 0;
+			}			
+		} catch (Exception e) {
+			logger.error("{} - Message processing failed: {} {}", LoggingUtil.getMethodName(), ret_code, System.lineSeparator());							
+			
+			// Depending on when the handshake process fails the ret_code might actually return HNET_RTRN_SUCCESS
+			// If this is the case use a HNET_RTRN_INVALIDFORMATERROR instead as it is likely a formatting error
+			if (ret_code.contentEquals(MessageUtil.HNET_RTRN_SUCCESS)) {
+				ret_code = MessageUtil.HNET_RTRN_INVALIDFORMATERROR;
+			}
+			hnSecureResponse = ErrorBuilder.buildDefaultErrorMessage(ret_code);
+			try {
+				sendResponse(socketOutput, hnSecureResponse);
+			} catch (IOException ioe) {
+				logger.error(ioe.getMessage());
```

Comment:
(not part of this commit) Syntax is not matching as other logs. Same comment for other logger.error(...)